pkg/skaffold/jib/jib.go

```diff
@@ -18,13 +18,76 @@ package jib
 
 import (
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha3"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/pkg/errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
 )
 
-func GetDependenciesMaven(_ /*workspace*/ string, _ /*a*/ *v1alpha3.JibMavenArtifact) ([]string, error) {
-	return nil, errors.New("jib maven support is unimplemented")
+// GetDependenciesMaven finds the source dependencies for the given jib-maven artifact.
+// All paths are absolute.
+// TODO(coollog): Add support for multi-module projects.
+func GetDependenciesMaven(workspace string, a *v1alpha3.JibMavenArtifact) ([]string, error) {
+	executable, subCommand := getCommandMaven(workspace, a)
+	return getDependencies(workspace, "pom.xml", executable, subCommand, "jib-maven")
 }
 
-func GetDependenciesGradle(_ /*workspace*/ string, _ /*a*/ *v1alpha3.JibGradleArtifact) ([]string, error) {
-	return nil, errors.New("jib gradle support is unimplemented")
+// GetDependenciesGradle finds the source dependencies for the given jib-gradle artifact.
+// All paths are absolute.
+// TODO(coollog): Add support for multi-module projects.
+func GetDependenciesGradle(workspace string, a *v1alpha3.JibGradleArtifact) ([]string, error) {
+	executable, subCommand := getCommandGradle(workspace, a)
+	return getDependencies(workspace, "build.gradle", executable, subCommand, "jib-gradle")
+}
+
+func exists(workspace string, filename string) bool {
+	_, err := os.Stat(filepath.Join(workspace, filename))
+	return err == nil
+}
+
+func getDependencies(workspace string, buildFile string, executable string, subCommand []string, artifactName string) ([]string, error) {
+	if !exists(workspace, buildFile) {
+		return nil, errors.Errorf("no %s found", buildFile)
+	}
+
+	cmd := exec.Command(executable, subCommand...)
+	cmd.Dir = workspace
+	stdout, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting %s dependencies", artifactName)
+	}
+
+	return getDepsFromStdout(string(stdout)), nil
+}
+
+const (
+	mavenExecutable  = "mvn"
+	gradleExecutable = "gradle"
+)
+
+func getCommandMaven(workspace string, a *v1alpha3.JibMavenArtifact) (executable string, subCommand []string) {
+	subCommand = []string{"jib:_skaffold-files", "-q"}
```

Comment:
`com.google.cloud.tools:jib-maven-plugin::_skaffold-files` — I discovered by accident that we need to use partially-qualified plugin references in case the plugin isn't configured in the pom.

Reply:
I believe currently we are only planning to support projects that have the plugin configured due to difficulties with multi-module projects if the plugin isn't configured @loosebazooka

Reply:
This is more of a concern in multimodule projects, where the processing needs to be multi module aware. But I would much prefer it was explicit.

Reply:
sorry, I meant explicitly defined in the pom.xml.